ClimsoftVer4/ClimsoftVer4/ucrFormDaily2.vb

```diff
@@ -285,4 +293,70 @@ Public Class ucrFormDaily2
+        Dim dtObsDateTime As Date
+        Dim lElementID As Long
+        Dim iPeriod As Integer
+
+        clsAllRecordsCall.SetTableName("form_daily2")
+        clsAllRecordsCall.SetFields(lstAllFields)
+        dtbAllRecords = clsAllRecordsCall.GetDataTable()
+
+        For Each row As DataRow In dtbAllRecords.Rows
+            For i As Integer = 1 To 31
+                rcdObservationInitial = Nothing
+                rcdObservationInitial = New observationinitial
+                If i < 10 Then
+                    strCurrTag = "0" & i
+                Else
+                    strCurrTag = i
+                End If
+                If Not IsDBNull(row.Item("day" & strCurrTag)) AndAlso Strings.Len(row.Item("day" & strCurrTag)) > 0 Then
+                    rcdObservationInitial.recordedFrom = row.Item("stationId")
+                    If Long.TryParse(row.Item("elementId"), lElementID) Then
+                        rcdObservationInitial.describedBy = lElementID
+                    Else
+                        Exit Sub
+                    End If
+                    Try
+                        dtObsDateTime = New Date(year:=row.Item("yyyy"), month:=row.Item("mm"), day:=i, hour:=row.Item("hh"), minute:=0, second:=0)
+                        rcdObservationInitial.obsDatetime = dtObsDateTime
+                    Catch ex As Exception
+
+                    End Try
+                    rcdObservationInitial.obsLevel = "surface"
+                    rcdObservationInitial.obsValue = row.Item("day" & strCurrTag)
+                    rcdObservationInitial.flag = row.Item("flag" & strCurrTag)
+                    If Integer.TryParse(row.Item("period" & strCurrTag), iPeriod) Then
+                        rcdObservationInitial.period = iPeriod
+                    End If
+                    rcdObservationInitial.qcStatus = 0
+                    rcdObservationInitial.acquisitionType = 1
+                    rcdObservationInitial.dataForm = "form_daily2"
+                    If Not IsDBNull(row.Item("signature")) Then
+                        rcdObservationInitial.capturedBy = row.Item("signature")
+                    End If
+                    If Not IsDBNull(row.Item("temperatureUnits")) Then
+                        rcdObservationInitial.temperatureUnits = row.Item("temperatureUnits")
+                    End If
+                    If Not IsDBNull(row.Item("precipUnits")) Then
+                        rcdObservationInitial.precipitationUnits = row.Item("precipUnits")
+                    End If
+                    If Not IsDBNull(row.Item("cloudHeightUnits")) Then
+                        rcdObservationInitial.cloudHeightUnits = row.Item("cloudHeightUnits")
+                    End If
+                    If Not IsDBNull(row.Item("visUnits")) Then
+                        rcdObservationInitial.visUnits = row.Item("visUnits")
+                    End If
+                    'clsDataConnection.db.observationinitials.Add(rcdObservationInitial)
```

Comment:
Note that this is commented as there is currently an issue with the database/model when uploading multiple records with the same station. So currently this doesn't do anything but will work once we resolve the issue with Ian.